tests/virtualization/universal/windows.pm

```diff
@@ -44,8 +49,18 @@ sub run {
     assert_script_run "ssh $username\@$_ 'systeminfo' | tee /tmp/$_-systeminfo.txt"                            foreach (keys %virt_autotest::common::imports);
     upload_logs "/tmp/$_-systeminfo.txt"                                                                       foreach (keys %virt_autotest::common::imports);
     assert_script_run "ssh $username\@$_ 'systeminfo' | grep '$virt_autotest::common::imports{$_}->{version}'" foreach (keys %virt_autotest::common::imports);
+}
 
-    remove_guest $_ foreach (keys %virt_autotest::common::imports);
+sub post_fail_hook {
+    my $self = shift;
+    # Note: Don't cleanup guests on test failure, so their state is preserved for debugging purposes!
```

Comment:
Note: This looks not necessary but I'd leave it here so that we know in the future, that one should not do a cleanup on test failure. So: No functional advantage but conservation of thought.